src/parser/expression.js

```diff
@@ -786,6 +795,8 @@ export default class ExpressionParser extends LValParser {
     node.callee = this.parseNoCallExpr();
     const optional = this.eat(tt.questionDot);
 
+    this.parseNewTypeArguments(node);
```

Comment:
I didn't notice this before: is it possible to implement new type arguments only in the typescript plugin?

Something like
```js
parseNew() {
    const node = super.parseNew();
    if (/* node is NewExpression */ && !/* node has args parentheses*/ && this.match("<")) {
       // parse type arguments
       // maybe parse arguments
    }
}
```
---

Anyway, apart from this detail this pr looks good to me! (well, I have to admit I haven't reviwed *all* of it because I don't know typescript very well 😅 )